src/UiPath.Workflow.Runtime/Expressions/CompiledExpressionInvoker.cs

```diff
@@ -40,7 +41,10 @@ public CompiledExpressionInvoker(ITextExpression expression, bool isReference, C
 
         _metadataRoot = metadata.Environment.Root;
 
-        ProcessLocationReferences();
+        if (!metadata.Environment.IsValidating)
```

Comment:
IsValidating is default false;
I can be set to true only in InternalActivityValidationServices constructor which is only called from Studio code either directly or indirectly. It is not called from anywhere in the production Robot code or Activities repository production code.
So the changes described in this PR do not affect the Robot.
In regards to the problem being fixed, this should be assessed by @aoltean16

Reply:
I've already asked for review from Alex, but as a general principle, this can also be reproduced in a simple unit test in this project (create a model item with 1400 variables and validate it with/without the ForceExpressionCache flag and observe the memory).

If we're talking about ownership of this code, it seems that the constructor of the public class CompiledExpressionInvoker does allocations which are unncessary in some cases.